### PR TITLE
Adding custom domains on infra

### PR DESCRIPTION
**Pull request by MiguelAngelGutierrezMaya**
### Core changes

- Added custom domain support to the v2 API Gateway stack, including ACM certificate provisioning, Route53 alias record creation, and base path mapping for a regional API Gateway domain.
- Wired `CUSTOM_DOMAIN`, `CUSTOM_DOMAIN_HOSTED_ZONE_ID`, and `API_CUSTOM_DOMAIN_PREFIX` into the v2 stack factory and exposed the custom API URL to Amplify so the frontend uses the custom endpoint when available.
- Updated the infrastructure deployment workflow to pass the custom domain secret and variables required for API Gateway domain provisioning.

### Schema changes

- None.

### Minor changes

- Added default placeholders for the new custom domain settings in `config/.env.defaults`.
- Added unit coverage for the custom domain flow in `api-gateway-stack.test.ts`, including the no-domain, prefixed-domain, and root-domain scenarios.
- Updated stack ordering comments to reflect the API Gateway custom domain dependency.

## Screenshots/Recordings

N/A. This PR only changes infrastructure configuration and tests.

## How to Test

1. Set `CUSTOM_DOMAIN`, `CUSTOM_DOMAIN_HOSTED_ZONE_ID`, and `API_CUSTOM_DOMAIN_PREFIX` with valid values for the target environment.
2. Run `pnpm --dir infra test -- api-gateway-stack.test.ts`.
   Expected result: the test suite passes, including the new custom domain scenarios.
3. Run `pnpm infra:cdk deploy --context stacks=ApiGateway`.
   Expected result: CDK creates or updates the ACM certificate, API Gateway custom domain, base path mapping, and Route53 alias record.
4. Run `pnpm infra:cdk deploy --context stacks=AmplifyHosting`.
   Expected result: the frontend receives the custom API URL through `API_URL` when the custom domain is configured.
5. Call the deployed endpoint through the custom domain, for example `curl -I https://<api-subdomain>.<domain>/<stage>/`.
   Expected result: the request resolves through the custom domain and returns an HTTP response from API Gateway.

## Checklist

- [x] The PR description includes clear instructions on how to test the implementation manually.

##### Added Tests?

- [x] yes
  - [x] Verified that this PR is not creating duplicate use cases
  - [x] All edge cases described in the acceptance criteria are covered, and tests have clear and descriptive names
- [ ] no, because they aren't needed
- [ ] no, because I need help

_Override these when needed._

```sh
pnpm --dir infra test -- api-gateway-stack.test.ts
pnpm infra:cdk deploy --context stacks=ApiGateway
pnpm infra:cdk deploy --context stacks=AmplifyHosting
curl -I https://<api-subdomain>.<domain>/<stage>/
```

### CODE REVIEW

File: infra/lib/versions/v2/api-gateway-stack.ts [hunk 1]
```diff
@@ -127,6 +145,54 @@ export class ApiGatewayStack extends BaseStack {
...
+      });
+
+      apiDomainName.addBasePathMapping(this.api, {
+        basePath: '',
+        stage: this.api.deploymentStage,
+      });
+
+      new ARecord(this, `${stackName}-ApiAliasRecord`, {
+        zone: hostedZone,
+        recordName: fullDomain,
```
[the remark this answers is not in the document]
Comment (MiguelAngelGutierrezMaya) [reply infra/lib/versions/v2/api-gateway-stack.ts:197]: AWS CDK already handles that

File: config/.env.defaults [hunk 2]
```diff
@@ -119,6 +119,9 @@ ALLOWED_ORIGINS=http://localhost:3000,http://localhost:3001
 # ── Api gateway — Configuration ──────────────────────
 #
 STAGE=dev
+CUSTOM_DOMAIN=example.com
+API_CUSTOM_DOMAIN_PREFIX=example
+CUSTOM_DOMAIN_HOSTED_ZONE_ID=123456789012
```
[the remark this answers is not in the document]
Comment (MiguelAngelGutierrezMaya) [reply config/.env.defaults:124]: This is only an example file
